src/libsyntax/parse/parser/item.rs

```diff
@@ -1741,14 +1741,25 @@ impl<'a> Parser<'a> {
     }
 
     fn report_invalid_macro_expansion_item(&self) {
+        let has_close_delim = self.sess.source_map()
+            .span_to_snippet(self.prev_span)
+            .map(|s| s.ends_with(")") || s.ends_with("]"))
```

Comment:
```suggestion
            .map_or(false, |s| s.ends_with(")") || s.ends_with("]"));
```

Reply:
ah yes. will change to this api.

Reply:
no such api...

will revert back.

Reply:
ah, that `map_err` on Result type.

Reply:
https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or ?

Reply:
```
.span_to_snippet(self.prev_span)
.ok()
.map_or(false, |s| ...)

Reply:
I think revert it back is better.
I would send a PR to add `Result::map_or` later.

Reply:
I've reverted it back. 

Thank you for the new API. :)